app/models/emoji.rb

```diff
@@ -81,13 +107,19 @@ def self.url_for(name)
   end
 
   def self.cache_key(name)
-    "#{name}:v#{EMOJI_VERSION}:#{Plugin::CustomEmoji.cache_key}"
+    "#{name}#{cache_postfix}"
+  end
+
+  def self.cache_postfix
+    ":v#{EMOJI_VERSION}:#{Plugin::CustomEmoji.cache_key}"
```

Comment:
This is existing code but something feels off here. `::CustomEmoji` has no part in the cache key here. If we delete a `::CustomEmoji`, our cache will still think it is valid because the cache key remains the same.

Reply:
AH I see how we're handling it. When we delete a `::CustomEmoji`, you have to call `Emoji.clear_cache` manually so this is handled.